tests/v2/validation/provisioning/clustertemplates/provisioning_cluster_template_test.go

```diff
@@ -0,0 +1,335 @@
+			Default:  "",
+			Type:     "string",
+		},
+		3: {
+			Variable: "rancherKubernetesEngineConfig.services.etcd.backupConfig.s3BackupConfig.accessKey",
+			Required: isRequiredQuestion,
+			Default:  "",
+			Type:     "string",
+		},
+		4: {
+			Variable: "rancherKubernetesEngineConfig.services.etcd.backupConfig.s3BackupConfig.secretKey",
+			Required: isRequiredQuestion,
+			Default:  "",
+			Type:     "string",
+		},
+		5: {
+			Variable: "rancherKubernetesEngineConfig.services.etcd.backupConfig.s3BackupConfig.endpoint",
+			Required: isRequiredQuestion,
+			Default:  "",
+			Type:     "string",
+		},
+		6: {
+			Variable: "rancherKubernetesEngineConfig.services.etcd.backupConfig.s3BackupConfig.region",
+			Required: isRequiredQuestion,
+			Default:  "",
+			Type:     "string",
+		},
+	}
+)
+
+type ClusterTemplateProvisioningTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	standardUserClient *rancher.Client
+	provisioningConfig *provisioninginput.Config
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+
+	c.provisioningConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, c.provisioningConfig)
+
+	client, err := rancher.NewClient("", testSession) //c.client
+	require.NoError(c.T(), err)
+
+	c.client = client
+
+	if c.provisioningConfig.RKE1KubernetesVersions == nil {
+		rke1Versions, err := kubernetesversions.ListRKE1AllVersions(c.client)
+		require.NoError(c.T(), err)
+
+		c.provisioningConfig.RKE1KubernetesVersions = []string{rke1Versions[len(rke1Versions)-1]}
+	}
+
+	if c.provisioningConfig.CNIs == nil {
+		c.provisioningConfig.CNIs = []string{cniCalico}
+	}
+
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) createRke1ClusterTemplateAndTemplateRevisions(clusterTemplateRevisionConfig clustertemplates.ClusterTemplateRevisionConfig, answers map[string]string, members []mgmt.Member) *clustertemplates.ClusterTemplateConfig {
+	clusterTemplate, err := rke1.CreateRkeTemplate(c.client, members)
+	require.NoError(c.T(), err)
+
+	clusterTemplateRevision, err := rke1.CreateRkeTemplateRevision(c.client, clusterTemplateRevisionConfig, clusterTemplate.ID)
+	require.NoError(c.T(), err)
+
+	clusterTemplateClusterConfig := new(clustertemplates.ClusterTemplateConfig)
+	clusterTemplateClusterConfig.RKE1TemplateID = clusterTemplate.ID
+	clusterTemplateClusterConfig.RKE1TemplateRevisionID = clusterTemplateRevision.ID
+
+	if answers != nil {
+		clusterTemplateClusterConfig.RKETemplateAnswers = &mgmt.Answer{
+			Values: answers,
+		}
+	}
+	return clusterTemplateClusterConfig
+}
+
+func (c *ClusterTemplateProvisioningTestSuite) testCreateRke1ClusterWithTemplateAndVerify(client *rancher.Client, clusterTempClusterConfig *rke1.ClusterTemplateConfig) *mgmt.Cluster {
```

Comment:
can these helpers at a minimum be moved to a new file in this package just to separate it from the tests themselves? Also could you please remove the word `test` from funcs that aren't Testing anything